cosinnus_message/rocket_chat.py

```diff
@@ -728,25 +730,36 @@ def users_update(self, user, request=None, force_user_update=False, update_passw
         profile = user.cosinnus_profile
         rocket_email = user_data.get('emails', [{}])[0].get('address', None)
         # rocket_mail_verified = user_data.get('emails', [{}])[0].get('verified', None)
-        rocket_username = self._get_unique_username(profile)
-        if not rocket_username:
-            return
-        if force_user_update or user_data.get('name') != rocket_username or rocket_email != profile.rocket_user_email:
+        rocket_username = user_data.get('username')
+        username_changed = rocket_username is None or self.has_username_changed(profile, rocket_username)
+        if username_changed:
+            rocket_username = self._get_unique_username(profile)
```

Comment:
4. After (possibly) changing the value of  `rocket_username` here, should we maybe re-evaluate `username_changed` again by running `username_changed = self.has_username_changed(profile, rocket_username)` again, just to be sure?

Reply:
I think this would be redundant. `_get_unique_username` possibly appends as `-x` if collisions were detected which is ignored by the `has_username_changed` function.
But If you have any consideration let's add it, more cation won't hurt.